src/semantic_hashing_demo/main.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict, List
+
+import numpy as np
+from openai import OpenAI
+
+client = OpenAI()
+
+
+def get_embedding(text: str, model="text-embedding-3-small"):
+    """Get the embedding vector of a given text with default OpenAI embedding small model.
+    Small embedding model: 1536 len of float values.
+    Large embedding model: 3072 len of float values.
+
+    Args:
+        text (str): search query/reviews/comments.
+        model (str, optional): supported embedding model. Defaults to "text-embedding-3-small".
+
+    Returns:
+        CreateEmbeddingResponse: list of float values.
+    """
+    text = text.replace("\n", " ")
+    return client.embeddings.create(input=[text], model=model).data[0].embedding
+
+
+def hash_vector(v: List[np.float64], nbits: np.uint16) -> str:
+    """LSH random projection hash function.
+    It is a simple hash function that takes a vector and returns a binary string of length nbits.
+
+    Args:
+        v (List[np.float64]): embedding vector.
+        nbits (np.uint16): no. of hyperplanes.
+
+    Returns:
+        str: A binary string of length nbits.
+    """
+    # create a set of 4 hyperplanes (normal ⟂), with 2 dimensions
+    plane_norms = np.random.rand(int(nbits), len(v)) - 0.5
```

Comment:
Will this create different random planes every time? If so, this will make the hashes non-comparable.

Reply:
Fixed this with seeded hyperplanes now. https://github.com/abhi3700/semantic-hashing-demo/commit/c42872b2224b2ef3f324842c322d237be77ef90d